generated/ruby/test/test_errorreporting.rb

```diff
@@ -0,0 +1,101 @@
+ReportedErrorEvent = Google::Devtools::Clouderrorreporting::V1beta1::ReportedErrorEvent
+ServiceContext = Google::Devtools::Clouderrorreporting::V1beta1::ServiceContext
+ErrorContext = Google::Devtools::Clouderrorreporting::V1beta1::ErrorContext
+HttpRequestContext = Google::Devtools::Clouderrorreporting::V1beta1::HttpRequestContext
+SourceLocation = Google::Devtools::Clouderrorreporting::V1beta1::SourceLocation
+
+report_errors_service_api = ReportErrorsServiceApi.new
+formatted_project_name = ReportErrorsServiceApi.project_path(PROJECT_NAME)
+http_request_context = HttpRequestContext.new(method: "GET", url: "https://20160920t135819-dot-gapic-test.appspot.com")
+source_location1 = SourceLocation.new(file_path: "foo/bar", line_number: 20, function_name: "doGet")
+source_location2 = SourceLocation.new(file_path: "foo/bar", line_number: 30, function_name: "doSet")
+p "Reporting error event 1"
+event1 = ReportedErrorEvent.new
+event1.service_context = ServiceContext.new(service: SERVICE, version: VERSION)
+event1.message = "error reporting test event1"
+event1.context = ErrorContext.new(http_request: http_request_context, user: USER_NAME, report_location: source_location1)
+report_errors_service_api.report_error_event(formatted_project_name, event1)
+p "Reporting error event 2"
+event2 = ReportedErrorEvent.new
+event2.service_context = ServiceContext.new(service: SERVICE, version: VERSION)
+event2.message = "error reporting test event2"
+event2.context = ErrorContext.new(http_request: http_request_context, user: USER_NAME, report_location: source_location1)
+report_errors_service_api.report_error_event(formatted_project_name, event2)
+p "Reporting error event 3"
+event3 = ReportedErrorEvent.new
+event3.service_context = ServiceContext.new(service: SERVICE, version: VERSION)
+event3.message = "error reporting test event3"
+event3.context = ErrorContext.new(http_request: http_request_context, user: USER_NAME, report_location: source_location2)
+report_errors_service_api.report_error_event(formatted_project_name, event3)
+
+#
+# Integration test for ErrorStatsServiceApi.list_group_stats, ErrorStatsServiceApi.list_events
+#
+require "google/cloud/error_reporting/v1beta1/error_stats_service_api"
+require "google/devtools/clouderrorreporting/v1beta1/error_stats_service_pb"
+ErrorStatsServiceApi = Google::Cloud::ErrorReporting::V1beta1::ErrorStatsServiceApi
+QueryTimeRange = Google::Devtools::Clouderrorreporting::V1beta1::QueryTimeRange
+
+p "Listing all the error groups"
+error_stats_service_api = ErrorStatsServiceApi.new
+formatted_project_name = ErrorStatsServiceApi.project_path(PROJECT_NAME)
+time_range = QueryTimeRange.new(period: :PERIOD_1_DAY)
+error_group_id = 0
+error_stats_service_api.list_group_stats(formatted_project_name, time_range).each_with_index do |error_group_stat, index|
+  p [error_group_stat.group.name, error_group_stat.group.group_id, error_group_stat.count].join(" ")
```

Comment:
I prefer to use a format instead of creating a style itself. It could be `puts error_group_stat.to_yaml` (format in YAML) or `puts JSON.pretty_generate(error_group_stat)` Same thing can be applied all among the file.